src/mount/mastercomm.cc

```diff
@@ -216,13 +212,32 @@ static inline void setDisconnect(bool value) {
 void fs_inc_acnt(uint32_t inode) {
 	std::unique_lock<std::mutex> acquiredFileLock(acquiredFileMutex);
 	acquiredFiles[inode]++;
+#ifdef _WIN32
+	if (gCleanAcquiredFilesPeriod > 0) {
+		if (acquiredFiles[inode] == 1) {
+			addAcquiredFileLastTimeUsed(inode, time(nullptr));
+		} else {
+			updateLastTimeUsedAcquireFile(inode, time(nullptr));
+		}
+	}
+#endif
 }
 
 void fs_dec_acnt(uint32_t inode) {
 	std::unique_lock<std::mutex> afLock(acquiredFileMutex);
+	if (!acquiredFiles.contains(inode)) {
+		return;
+	}
 	auto &cnt = acquiredFiles[inode];
 	cnt--;
-	if (cnt <= 0) { acquiredFiles.erase(inode); }
+	if (cnt <= 0) {
+#ifdef _WIN32
+		if (gCleanAcquiredFilesPeriod > 0) {
+			removeAcquiredFileLastTimeUsed(inode);
```

Comment:
```suggestion
			removeAcquiredFile(inode);
```

Reply:
Considered and commented [here ](https://github.com/leil-io/saunafs/pull/230#discussion_r1872378478)